schemas/sublime-completions.json

```diff
@@ -32,11 +56,44 @@
               },
               "contents": {
                 "type": "string",
-                "markdownDescription": "A text to be inserted when completion is triggered. Supports the same string [interpolation features](https://sublime-text-unofficial-documentation.readthedocs.io/en/latest/extensibility/snippets.html#snippet-features) as snippets.\n\n**Note**: If you want a literal $, you have to escape it like this: `\\$`."
+                "markdownDescription": "**Completion Contents**\n\nA text to be inserted when completion is committed.\n\n_Notes:_\n\n- Supports the same string [interpolation features](https://www.sublimetext.com/docs/completions.html#snippet_variables) as snippets.\n- Literal $ need to be escaped like this: `\\$`."
+              },
+              "annotation": {
+                "type": "string",
+                "markdownDescription": "**Completion Annotation**\n\nAnnotations are displayed on the right-hand edge of the completions popup,\nand may contain any information the author deems useful.\nTypically the annotations will be a word or two."
+              },
+              "details": {
+                "type": "string",
+                "markdownDescription": "**Completion Details**\n\nA single line description of the completion.\nMay contain the following HTML tags for basic formatting:\n\n_Example:_\n\n```js\n\"details\": \"Wraps selection in a <code>&lt;b&gt;</code> tag\"\n```"
+              },
+              "kind": {
+                "markdownDescription": "**Completion Kind Metadata**\n\nThe kind metadata for the completion.\n\n---\n\n**String**\n\nIf the value is a string, it must be one of `ambiguous`, `function`, `keyword`, `markup`, `namespace`,\n`navigation`, `snippet`, `type`, `variable`.\n\n_Example:_\n\n```js\n\"kind\": \"function\"\n```\n\n---\n\n**Array**\n\nIf the value is a 3-element array of strings, they must be:\n\n1. A string from the list above, which is used by the theme to select the color of the kind metadata\n2. A single Unicode character to be shown to the left of the trigger\n3. A description of the kind, viewable in the kind letter tool tip, and the detail pane (when visible)\n\n_Example:_\n\n```js\n\"kind\": [\"function\", \"m\", \"Method\"]\n```",
+                "oneOf": [
+                    {
+                      "type": "array",
+                      "items": [
+                        {
+                          "markdownDescription": "**Kind Category**\n\nA highlevel category of a symbol kind,\nwhich is used by the theme to select the color of the kind metadata.\n\nIt must be one of: \n\n- `\"ambiguous\"`\n- `\"function\"`\n- `\"keyword\"`\n- `\"markup\"`\n- `\"namespace\"`\n- `\"navigation\"`\n- `\"snippet\"`\n- `\"type\"`\n- `\"variable\"`",
```

Comment:
- Maybe add a space in "highlevel"
- Can we skip the double quotes around category types to make it look cleaner?

Reply:
I'll use `high-level` as it is the term in the official completions docs.

Reply:
Yes, sorry, that's even better.